pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -3229,21 +3230,36 @@ impl Timeline {
             }
             return Ok(CompactionOutcome::Done);
         }
-        self.compact_with_gc_inner(cancel, job, ctx, yield_for_l0)
+        self.compact_with_gc_inner(cancel, job, ctx, yield_for_l0, do_metadata_compaction)
             .await
     }
 
     async fn compact_with_gc_inner(
         self: &Arc<Self>,
         cancel: &CancellationToken,
-        job: GcCompactJob,
+        mut job: GcCompactJob,
         ctx: &RequestContext,
         yield_for_l0: bool,
+        do_metadata_compaction: bool,
     ) -> Result<CompactionOutcome, CompactionError> {
         // Block other compaction/GC tasks from running for now. GC-compaction could run along
         // with legacy compaction tasks in the future. Always ensure the lock order is compaction -> gc.
         // Note that we already acquired the compaction lock when the outer `compact` function gets called.
 
+        if !do_metadata_compaction && job.compact_key_range.end > Key::metadata_key_range().start {
```

Comment:
Won't the
```
job.compact_key_range.end > Key::metadata_key_range().start
```
condition be true for any job that is completely _after_ (think: right of) the metadata key range?
If I'm right about that we'd
- be making log noise with this `info!` here and worse
- adjust the job range end to somewhere before the range start.

We do have `range_overlaps()` to do the correct check.

---

Or, what am I missing here?

Reply:
I think this is equivalent to "key range overlaps or is completely right to the metadata key range"? Metadata key covers 0x60-0xFF key prefixes. But we can do range_overlaps here.